internal/store/static.go

```diff
@@ -150,6 +155,25 @@ func (store *StaticStore) getPlugins(serverVersion string, includeEnterprisePlug
 			}
 		}
 
+		if platform != "" {
+			var bundle model.PlatformBundleMetadata
+			switch platform {
+			case model.LinuxAmd64:
+				bundle = storePlugin.Platforms.LinuxAmd64
+			case model.DarwinAmd64:
+				bundle = storePlugin.Platforms.DarwinAmd64
+			case model.WindowsAmd64:
+				bundle = storePlugin.Platforms.WindowsAmd64
+			}
+
+			if bundle.DownloadURL != "" && bundle.Signature != "" {
+				newRef := *storePlugin
+				newRef.DownloadURL = bundle.DownloadURL
+				newRef.Signature = bundle.Signature
+				storePlugin = &newRef
```

Comment:
Can you please elaborate why the pointer shenanigans are needed?

Reply:
@hanzei I replied on the other conversation https://github.com/mattermost/mattermost-marketplace/pull/116#discussion_r510591763 

When the server runs locally, it caches the `plugins.json` contents in memory. When the pointer of the struct is edited here (without my hacky fix here), it edits the master copy that is stored in memory. Maybe we should just not use pointers to reference the source data.

Reply:
@hanzei If this is something that needs to be addressed, do you mind if I make it a follow up ticket so we can merge this?

I've tried the method of using `[]model.Plugin` instead of `[]*model.Plugin` to avoid mutation of the source data, but a lot of different structs here implement the interface that has the method's return value as `[]*model.Plugin`. If you think avoiding pointers altogether is the right solution, 2/5 I think we should offload this change to another ticket.

Reply:
Surely we can merge this PR as it is. I didn't intended to block QA review.